src/charm.py

```diff
@@ -117,6 +125,123 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
+
+    def _has_secrets(self) -> bool:
+        """Check if current Juju version supports secrets.
+
+        Returns:
+            If secrets are supported or not.
+        """
+        juju_version = JujuVersion.from_environ()
+        # Because we're only using secrets in a peer relation we don't need to
+        # check if the other end of a relation also supports secrets.
+        return juju_version.has_secrets
```

Comment:
I re-used the code from here:
https://github.com/canonical/indico-operator/pull/211
I think having a _has_secrets it's easier to understand and it could be changed if there is no JujuVersion.from_envriron anymore.

Reply:
At first, I answered that I preferred the _has_secrets in case of JujuVersion changes but then while working on Mjolnir module + collect_unit_status I realized that your suggestion is better since I don't need to import anything from the charm itself. So: changed :)